src/include/wiredtiger.in

```diff
@@ -1261,6 +1261,15 @@ struct __wt_session {
 	    const char *uri, const char *newuri, const char *config);
 
 	/*!
+	 * Reset the session. Any resources held by the session are released.
+	 * @snippet ex_all.c Reset the session
```

Comment:
I also think it'd would be good to have more information, what resources are released, that an error will be generated if there is a transaction running, what happens if there are open cursors.